bundles/AdminBundle/Session/Handler/AdminSessionHandler.php

```diff
@@ -98,7 +108,15 @@ public function useSessionAttributeBag(callable $callable, string $name = 'pimco
 
         $result = call_user_func_array($callable, [$attributeBag, $session]);
 
-        $this->writeClose();
+        // write & close session when in admin context
+        // https://github.com/pimcore/pimcore/pull/12022#issuecomment-1119451897
+        $request = $this->requestHelper->getCurrentRequest();
+        if (
+            $this->matchesPimcoreContext($request, PimcoreContextResolver::CONTEXT_ADMIN) &&
```

Comment:
```suggestion
            $this->matchesPimcoreContext($request, PimcoreContextResolver::CONTEXT_ADMIN) ||
```
This should be an `or` as the context it not admin in case of `isFrontendRequestByAdmin() === true`
See e.g. https://github.com/pimcore/pimcore/blob/d124b16a837567ed9b7ff010fdeff6e82e9bfaf3/bundles/AdminBundle/EventListener/AdminSecurityListener.php#L70

I'm also just thinking whether we should do the check on the main request and cache the result for further use? wdyt?

Reply:
Changed to `||` check and added `$canWriteAndClose` property for caching the result for subsequent requests.